### PR TITLE
fix(MongoConnection) - Sorts List and Set DTO's by inserted timestamp

### LINKED ISSUE #74: Logs are returned in the wrong order when using Hangfire.Console with Hangfire.Mongo
When using Hangfire.Console, the messages are returned in no particular order. See relevant discussion on Hangfire.Console : https://github.com/pieceofsummer/Hangfire.Console/issues/36

When using PerformContext.WriteLine, the logs are often in the wrong order.

console-wrong-log-order

Current Setup : 

- Hangfire 1.6.14
- Hangfire.Console 1.3.3
- Hangfire.Mongo 0.4.0 
- Mongo Version : 3.0.12 on Windows

It seems okay when the first log appears, however when the job is finished it gets moved to the end of the paragraph.

It used to be on any log, however I can't seem to reproduce at the moment, now it's usually the first log that changes position.

### PR DESCRIPTION
**Pull request by gottscj**
Fixes #74

### CODE REVIEW

File: src/Hangfire.Mongo/MongoConnection.cs [hunk 1]
```diff
@@ -500,6 +503,7 @@ public override List<string> GetRangeFromList(string key, int startingFrom, int
                 .StateData
                 .OfType<ListDto>()
                 .Find(Builders<ListDto>.Filter.Eq(_ => _.Key, key))
+                .SortBy(_ => _.Id)
```
Comment (pieceofsummer) [inline src/Hangfire.Mongo/MongoConnection.cs]: Should be `.SortByDescending(_ => _.Id)`

File: src/Hangfire.Mongo/MongoConnection.cs [hunk 2]
```diff
@@ -515,6 +519,7 @@ public override List<string> GetAllItemsFromList(string key)
...
```
Comment (pieceofsummer) [inline src/Hangfire.Mongo/MongoConnection.cs]: Should be `.SortByDescending(_ => _.Id)`

File: src/Hangfire.Mongo/MongoConnection.cs (hunk 1 above)
Comment (gottscj) [inline src/Hangfire.Mongo/MongoConnection.cs]: Thanks! Only looked at the "Set" part in "SqlServerConnection.cs"

File: src/Hangfire.Mongo/MongoConnection.cs (hunk 2 above)
Comment (gottscj) [inline src/Hangfire.Mongo/MongoConnection.cs]: Thanks!

**Pull request merged**